core/app/models/spree/taxon.rb

```diff
@@ -85,6 +85,19 @@ def active_products
       products.not_deleted.available
     end
 
+    # @return [ActiveRecord::Relation<Spree::Product>] all self and descendant products
+    def all_products
+      scope = Product.joins(:taxons)
+      scope.where(
+        spree_taxons: { id: descendants.select(:id) }
+      ).or(scope.where(spree_taxons: { id: id } ))
```

Comment:
This could probably be simpler as `where(spree_taxons: { id: self_and_descendants.ids } )`